pkg/health/checker/checker.go

```diff
@@ -45,7 +51,35 @@ type HealthChecker interface {
 }
 
 type ShadowTrafficHealthChecker interface {
+	WatchPodOnNode(
+		ctx context.Context,
+		nodeID types.NodeName,
+		podID types.PodID,
+		useHealthService bool,
+		useOnlyHealthService bool,
+		status manifest.StatusStanza,
+	) (chan health.Result, chan error)
 	WatchService(
+		ctx context.Context,
```

Comment:
I thought the idea of the ShadowTrafficHealthChecker was that it would implement the same interface as the HealthChecker. How is the migration off of this ShadowTrafficHealthChecker suppose to proceed?

If you need a different type to support health checks for daemon sets, it's ok to make a new type. I'd like this interface to maintain similarity with the HealthChecker interface.

Reply:
So ShadowTrafficHealthChecker should check all nodes for WatchService and Service and the checker type for daemon sets could only check DS nodes for WatchService and Service.

Reply:
For the migration I was thinking that everything will eventually be using the shadowChecker. And when we want to finalize the migration, the consul code and health service flags will be removed and the shadowChecker will be renamed.